src/main.rs

```diff
@@ -51,3 +108,57 @@ fn listen() {
         }
     }
 }
+
+/// Creates a Config struct out of a config file
+/// returns default values for everything that is
+/// not entered manually
+fn read_conf_from_json(name: String) -> Config {
+
+    #[derive(Debug, RustcDecodable, Default)]
+    struct CfgFile {
+        address: Option<String>,
+        port: Option<u16>,
+        dir: Option<String>
+    }
+
+    // Read from JSON file and decode to CfgFile
+    let mut config = CfgFile::default();
+    if let Ok(mut f) = File::open(name) {
+        let mut s = String::new();
+        if let Err(e) = f.read_to_string(&mut s) {
+            error!("Could not read JSON-file: {:?}", e)
+        } else {
+            config = json::decode(&s).unwrap();
+        }
+    }
+
+    // Return configuration, all None datafields set to default
+    Config {
+        address: string_to_ipv4(config.address.unwrap_or("127.0.0.1".into())),
+        port: config.port.unwrap_or(4242),
+        dir: config.dir.unwrap_or("data".into())
+    }
+}
+
+/// Converts String into Ipv4Addr
+fn string_to_ipv4(s : String) -> Ipv4Addr {
+    let ip_parts : Vec<&str> = s.split(".").collect();
+
+    let mut part_convert : Vec<u8> = Vec::default();
+    for s in ip_parts {
+        match s.parse::<u8>() {
+            Ok(n) => part_convert.push(n),
+            Err(e) => error!("Parsing error: {:?}", e)
+        };
+    }
+    Ipv4Addr::new(part_convert[0], part_convert[1],
+                  part_convert[2], part_convert[3])
+}
```

Comment:
There is already a method that does that. It's called `Ipv4Addr::from_str()` (from the trait `FromStr` which `Ipv4Addr` implements). Please use that instead of this. Sorry ;)